OutOfSchool/OutOfSchool.WebApi/Services/ParentService.cs

```diff
@@ -52,19 +55,25 @@ public async Task Delete(Guid id)
 
         await currentUserService.UserHasRights(new ParentRights(id));
 
-        var entity = new Parent() {Id = id};
+        var entity = await repositoryParent.GetById(id).ConfigureAwait(false);
 
-        try
+        if (entity is null)
+        {
+            var message = $"Parent with Id = {id} doesn't exist in the system.";
+            logger.LogError(message);
+            throw new ArgumentException(message, nameof(id));
+        }
+
+        await repositoryParent.RunInTransaction(async () =>
         {
             await repositoryParent.Delete(entity).ConfigureAwait(false);
 
             logger.LogInformation("Parent with Id = {Id} successfully deleted", id);
-        }
-        catch (DbUpdateConcurrencyException)
-        {
-            logger.LogError("Deleting failed. Parent with Id = {Id} doesn't exist in the system", id);
-            throw;
-        }
+
+            await userService.Delete(entity.UserId);
+
+            logger.LogInformation("User with Id = {UserId} successfully deleted", entity.UserId);
+        });
```

Comment:
(Logic, logs) There is potential of having inconsistent logs. Transaction could be rollbacked after execution of this lambda. As a result you could have logs stating that "X successfully deleted", but nothing changed in db. 
Please consider logging after the `await ...RunInTransaction(...)`. 

(Concurrency) `.ConfigureAwait()` is inconsistently used, either use it for both awaitables or remove it for both.

Reply:
> (Logic, logs) There is potential of having inconsistent logs. Transaction could be rollbacked after execution of this lambda. As a result you could have logs stating that "X successfully deleted", but nothing changed in db. Please consider logging after the `await ...RunInTransaction(...)`.
> 
> (Concurrency) `.ConfigureAwait()` is inconsistently used, either use it for both awaitables or remove it for both.

Sorry for interrupting, but one more point regarding logs: While I believe that `userService` may already include some logging logic, even if it doesn’t, in my opinion, `ParentService` should not be responsible for logging information related to user removal status.

Reply:
Done